File: src/Case.Subs.KeyMatcher/Case.Subs.KeyMatcher.SharedProject/Entry/CmdMain.cs
using System;
using Autodesk.Revit.Attributes;
using Autodesk.Revit.DB;
using Autodesk.Revit.UI;
using Case.Subs.KeyMatcher.Utility;

namespace Case.Subs.KeyMatcher.Entry
{

  [Transaction(TransactionMode.Manual)]
  public class CmdMain : IExternalCommand, IExternalCommandAvailability
  {

    /// <summary>
    /// Report Groups by View
    /// </summary>
    /// <param name="commandData"></param>
    /// <param name="message"></param>
    /// <param name="elements"></param>
    /// <returns></returns>
    public Result Execute(ExternalCommandData commandData, ref string message, ElementSet elements)
    {

      try
      {
          clsGlobals.RevitYear = commandData.Application.Application.VersionNumber.ToString();
          // Success
        return Result.Succeeded;

      }
      catch (Exception ex)
      {

        // Failure
        message = ex.Message;
        return Result.Failed;

      }

    }

    /// <summary>
    /// Command Availability
    /// </summary>
    /// <param name="applicationData"></param>
    /// <param name="selectedCategories"></param>
    /// <returns></returns>
    public bool IsCommandAvailable(UIApplication applicationData, CategorySet selectedCategories)
    {
      return applicationData.ActiveUIDocument != null;
    }

  }
}

File: src/Case.ViewportReporting/Case.ViewportReporting.SharedProject/Entry/CmdMain.vb
Imports Autodesk.Revit.Attributes
Imports Autodesk.Revit.DB
Imports Autodesk.Revit.UI
Imports [Case].ViewportReporting.Data

Namespace Entry

  ''' <summary>
  ''' Revit 2021 Command Class
  ''' </summary>
  ''' <remarks></remarks>
  <Transaction(TransactionMode.Manual)>
  Public Class CmdMain

    Implements IExternalCommand

    ''' <summary>
    ''' Command
    ''' </summary>
    ''' <param name="commandData"></param>
    ''' <param name="message"></param>
    ''' <param name="elements"></param>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Function Execute(ByVal commandData As ExternalCommandData,
                            ByRef message As String,
                            ByVal elements As ElementSet) As Result Implements IExternalCommand.Execute
      Try
        ' Construct and Display the Form
        Dim d As New form_Main(New clsSettings(commandData, elements))
        d.ShowDialog()

        ' Return Success
        Return Result.Succeeded

      Catch ex As Exception

        ' Failure Message
        message = ex.Message
        Return Result.Failed

      End Try

    End Function

  End Class
End Namespace